File: backend/src/main/java/com/anderk222/hire/flow/interview/domain/model/Interview.java
package com.anderk222.hire.flow.interview.domain.model;

import com.anderk222.hire.flow.candidacy.domain.model.CandidacyId;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.List;

/**
 * Entidad de dominio para una entrevista agendada. `locationOrLink` es un
 * único campo a propósito: cuál es el significativo depende de {@link #mode}
 * (FA en CU-05: virtual lleva un link, presencial lleva una dirección/sala).
 */
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Interview {

    private InterviewId id;
    private CandidacyId candidacyId;
    private InterviewType type;
    private InterviewMode mode;
    private Instant scheduledAt;
    private String locationOrLink;
    private List<Long> participantIds;

    /**
     * Regla de negocio: ningún participante puede tener un conflicto de agenda (FE en CU-05).
     * {@code conflictingInterviews} ya viene resuelto por el caso de uso (consultando
     * {@link com.anderk222.hire.flow.interview.domain.InterviewRepository}); el modelo
     * no conoce la BD, solo decide si esos datos violan la regla.
     */
    public void checkParticipantsAvailability(List<Interview> conflictingInterviews) {
        // TODO: si algún participantId se repite entre conflictingInterviews y este.participantIds,
        // throw new SchedulingConflictException(participantId).
        throw new UnsupportedOperationException("TODO: implement availability check");
    }
}

File: frontend/src/app/features/interview/models/interview.ts
export type InterviewType = 'HR' | 'TECHNICAL' | 'HIRING_MANAGER';
export type InterviewMode = 'ON_SITE' | 'VIRTUAL';

/** Mirrors InterviewResponse from the backend `interview` module. */
export interface Interview {
  id: number;
  candidacyId: number;
  type: InterviewType;
  mode: InterviewMode;
  scheduledAt: string;
  locationOrLink: string;
  participantIds: number[];
}

/** Mirrors ScheduleInterviewCommand (UC-05). */
export interface ScheduleInterviewRequest {
  candidacyId: number;
  type: InterviewType;
  mode: InterviewMode;
  scheduledAt: string;
  locationOrLink: string;
  participantIds: number[];
  reminderEnabled: boolean;
}
